src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.workspace.integration;
+
+import bio.terra.workspace.app.Main;
+import bio.terra.workspace.generated.model.CreateWorkspaceRequestBody;
+import bio.terra.workspace.generated.model.CreatedWorkspace;
+import bio.terra.workspace.integration.common.configuration.TestConfiguration;
+import bio.terra.workspace.integration.common.response.WorkspaceResponse;
+import bio.terra.workspace.integration.common.utils.TestUtils;
+import bio.terra.workspace.integration.common.utils.WorkspaceManagerTestClient;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@Tag("integration")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class WorkspaceIntegrationTest {
+
+  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
+
+  @Autowired private TestUtils testUtils;
+
+  @Autowired private TestConfiguration testConfig;
+
+  @BeforeEach
+  public void setup() {}
+
+  @Test
+  public void createWorkspace() throws Exception {
+    UUID workspaceId = UUID.randomUUID();
+    String path = testConfig.getCreateWorkspaceUrlDev();
+    String userEmail = testConfig.getServiceAccountEmail();
+    CreateWorkspaceRequestBody body =
+        new CreateWorkspaceRequestBody()
+            .id(workspaceId)
+            .authToken("invalidToken")
+            .spendProfile(null)
+            .policies(null);
+    String json = testUtils.mapToJson(body);
+
+    WorkspaceResponse<CreatedWorkspace> workspaceResponse =
+        workspaceManagerTestClient.post(userEmail, path, json, CreatedWorkspace.class);
+
+    Assertions.assertEquals(workspaceResponse.getStatusCode(), HttpStatus.OK);
+    Assertions.assertNotNull(workspaceResponse.getResponseObject());
```

Comment:
Ah good call! The value can be `Optional.empty()` but not null. 
I'll add `Assertions.assertTrue(workspaceResponse.getResponseObject().isPresent());`
That removes the syntax warning on `.get()`. However, I don't think `.get()` itself can be safely removed though, because `workspaceResponse.getResponseObject()` alone will return an `Optional` object that's not cast-able to a `CreatedWorkspace` object. But let me know if I'm missing something.

Reply:
that's where the `orElse` in my second suggestion comes in handy ... `workspaceResponse.getResponseObject().orElse(null)` will get you a `CreatedWorkspace` object, which could be null.

Reply:
I used your second suggestion.